### PR TITLE
Add a Jupyter Notebook that downloads all categorized runs from recount2

### PR DESCRIPTION
**Pull request by ben-heil**
I started with a script from multiplier to download the recount2 data, then modified it to work for my dataset. As I was writing the script, I realized that it made sense to save the runs to different dataframes depdending on whether they will be fed into PLIER downstream or not, and whether the run represents a disease state or not.

### CODE REVIEW

File: brdnet/download_categorized_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,4362 @@
...
+   "source": [
+    "import random\n",
+    "import xml.etree.ElementTree as ET\n",
+    "\n",
+    "import pandas\n",
+    "from plotnine import *\n",
+    "import requests\n",
+    "from tqdm import tqdm_notebook\n",
+    "\n",
+    "%load_ext rpy2.ipython"
```
Comment (zietzm) [inline brdnet/download_categorized_data.ipynb:17]: Is this what allows you to use `%%R`? Very cool, didn't know that was even possible

File: brdnet/download_categorized_data.ipynb [hunk 2]
```diff
@@ -0,0 +1,4362 @@
...
+    "for accession in tqdm_notebook(study_accessions):\n",
+    "    data = requests.get('http://metasra.biostat.wisc.edu/api/v01/samples.json?study={}'.format(accession)).json()\n",
+    "    \n",
+    "    total_samples += data['sampleCount']\n",
+    "    \n",
+    "    for term in data['terms']:\n",
+    "        term_name = term['dterm']['name']\n",
+    "        term_ids = term['dterm']['ids']\n",
+    "        \n",
+    "        disease = False\n",
```
Comment (zietzm) [inline brdnet/download_categorized_data.ipynb]: https://github.com/greenelab/brd-net/blob/d4ce291992ae70b83f6101abad774905c6a70711/brdnet/download_categorized_data.ipynb#L318-L327

Minor suggestion, but above code could be replaced with the following--which is more succinct--or something similar:
```python
disease_id = next((id_ for id_ in term_ids if 'DOID' in id_), None)
if disease_id is None:
    continue
```
Note that the second argument in `next` is the default value, assuming that no value is given by the generator.

File: brdnet/download_categorized_data.ipynb [hunk 3]
```diff
@@ -0,0 +1,4362 @@
...
+    "        for id_ in term_ids:\n",
+    "            if 'DOID' in id_:\n",
+    "                disease_id = id_\n",
+    "                disease = True\n",
+    "                break\n",
+    "                \n",
+    "        if not disease:\n",
+    "            continue\n",
+    "        \n",
+    "        if disease_id in id_to_studies:\n",
```
Comment (zietzm) [inline brdnet/download_categorized_data.ipynb]: https://github.com/greenelab/brd-net/blob/d4ce291992ae70b83f6101abad774905c6a70711/brdnet/download_categorized_data.ipynb#L329-L334

Similar suggestion to above: not necessary but more succinct. Up to you; feel free to ignore if you have a strong preference one way or the other.

```python
id_to_studies[disease_id] = id_to_studies.setdefault(disease_id, 0) + 1
id_to_samples[disease_id] = id_to_samples.setdefault(disease_id, 0) + term['sampleCount']
```

File: brdnet/download_categorized_data.ipynb [hunk 4]
```diff
@@ -0,0 +1,4362 @@
...
+    "            id_to_studies[disease_id] += 1\n",
+    "            id_to_samples[disease_id] += term['sampleCount']\n",
+    "        else:\n",
+    "            id_to_studies[disease_id] = 1\n",
+    "            id_to_samples[disease_id] = term['sampleCount']\n",
+    "            id_to_name[disease_id] = term_name            "
```
Comment (zietzm) [inline brdnet/download_categorized_data.ipynb]: Will there be a 1 - 1 map between `disease_id` and `term_name`, or are there potentially multiple `term_name`s for a single `disease_id` and you want only to take the first one encountered?

File: brdnet/download_categorized_data.ipynb [hunk 5]
```diff
@@ -0,0 +1,4362 @@
...
+      "/home/ben/anaconda3/envs/brdnet/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: trying URL 'http://duffel.rail.bio/recount/v2/SRP064138/rse_gene.Rdata'\n",
+      "\n",
+      "  warnings.warn(x, RRuntimeWarning)\n",
+      "/home/ben/anaconda3/envs/brdnet/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning:  length 2921751 bytes (2.8 MB)\n",
+      "\n",
+      "  warnings.warn(x, RRuntimeWarning)\n",
+      "/home/ben/anaconda3/envs/brdnet/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: 2019-07-18 14:18:59 downloading file rse_gene.Rdata to ../data/recount/SRP041531\n",
+      "\n",
+      "  warnings.warn(x, RRuntimeWarning)\n",
+      "/home/ben/anaconda3/envs/brdnet/lib/python3.7/site-packages/rpy2/rinterface/__init__.py:146: RRuntimeWarning: trying URL 'http://duffel.rail.bio/recount/v2/SRP041531
```
Comment (zietzm) [inline brdnet/download_categorized_data.ipynb:4241]: Does this somehow pass the Python list to R?

File: brdnet/download_categorized_data.ipynb (hunk 1 above)
Comment (ben-heil) [reply brdnet/download_categorized_data.ipynb:17]: Yep, installing rpy2 lets you use R and python cells in the same notebook!

File: brdnet/download_categorized_data.ipynb (hunk 5 above)
Comment (ben-heil) [reply brdnet/download_categorized_data.ipynb:4241]: Yep, rpy lets you pass lists and dataframes between R and python

File: brdnet/download_categorized_data.ipynb (hunk 2 above)
Comment (ben-heil) [reply brdnet/download_categorized_data.ipynb]: That's a cool construct, will add it

File: brdnet/download_categorized_data.ipynb (hunk 3 above)
Comment (ben-heil) [reply brdnet/download_categorized_data.ipynb]: I like yours better, I'd never heard of the setdefault function before

File: brdnet/download_categorized_data.ipynb (hunk 4 above)
Comment (ben-heil) [reply brdnet/download_categorized_data.ipynb]: The mapping _should_ be 1 - 1, but nothing would go wrong if it was 1 - many and a random name was chosen. The Disease Ontology id is the unique identifier, while the name just makes it human readable.

**Pull request merged**